lib/decidim/cdtb/spam/user_spam_detector.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module Decidim
+  module Cdtb
+    module Spam
+      # Detect spam behavior in users
+      #
+      class UserSpamDetector < ::Decidim::Cdtb::Task
+        DEFAULT_SPAM_WORDS = %w[viagra sex game free crypto crack xxx luck girls vip download].freeze
+
+        def initialize(host = nil)
+          @host = host
+          progress_bar = { title: "Decidim::User" }
+          super("SPAM DETECTOR", progress_bar: progress_bar)
+        end
+
+        def prepare_execution(_ctx)
+          if @host.present?
+            @query = Decidim::Organization.where("host ilike ?", "%#{@host}%")
+            @users = Decidim::User.where(organization: @query.first)
+          else
+            @users = Decidim::User.all
+          end
```

Comment:
```suggestion
          @query = Decidim::Organization
          @query = @query.where("host ilike ?", "%#{@host}%") if @organization
          @users = @query.all
```

Reply:
But here `@query.all` is organizations, not users like in my code.